op-challenger/game/fault/preimages/large.go

```diff
@@ -74,6 +143,19 @@ func (p *LargePreimageUploader) initLargePreimage(ctx context.Context, uuid *big
 	return nil
 }
 
+// addLargePreimageLeafs adds leafs to the large preimage proposal.
+// This method *must* be called after calling [initLargePreimage].
+func (p *LargePreimageUploader) addLargePreimageLeafs(ctx context.Context, uuid *big.Int, input []byte, commitments [][32]byte, finalize bool) error {
+	tx, err := p.contract.AddLeaves(uuid, input, commitments, finalize)
+	if err != nil {
+		return fmt.Errorf("failed to create pre-image oracle tx: %w", err)
+	}
+	if err := p.sendTxAndWait(ctx, tx); err != nil {
+		return fmt.Errorf("failed to populate pre-image oracle: %w", err)
+	}
```

Comment:
Assuming the contract will bail out if tx arrive out of order, I suspect we want to send all the tx, then wait for the receipts. Otherwise we'll take forever to get them all submitted.  Might need to utilise the queue function in the txmgr to handle this properly though. https://github.com/ethereum-optimism/optimism/blob/87f50eb513049a050560b4594b1e7ab334ae3a93/op-service/txmgr/queue.go#L34

Reply:
We could do that in a follow up though.